internal/provider/resource_application.go

```diff
@@ -137,6 +137,13 @@ func (r *ResourceApplication) Read(ctx context.Context, req resource.ReadRequest
 		return
 	}
 
+	//this code should be 404, bug in the API
+	if httpResp.StatusCode() == 403 {
```

Comment:
I acting on a 403 here isn't good as that would also mean that people without actual permissions to remove the app, might accidentally remove it from the state.

Could you change this to a 404 and create a bug ticket for the API instead?

Reply:
Yup created last saturday WAL-6614

Reply:
so I think it's ok to wait until that is fixed

Reply:
If you change this to a 404, we can already merge this so it eventually works once the API is fixed :-)

Reply:
done